src/backend/distributed/executor/multi_executor.c

```diff
@@ -52,6 +52,19 @@ multi_ExecutorStart(QueryDesc *queryDesc, int eflags)
 		MultiExecutorType executorType = MULTI_EXECUTOR_INVALID_FIRST;
 		Job *workerJob = multiPlan->workerJob;
 
+		/*
+		 * It's possible - particularly in combination with prepared
+		 * statements - that no usable plan has been found. In that case error
+		 * contains the reason for that - raise it.  The reason for
+		 * conditionally erroring out is a slightly hacky approach to
+		 * supporting prepared statements. See
+		 * multi_planner.c:CreateDistributedPlan() for details.
+		 */
```

Comment:
I think it's worth elaborating / being more to the point here, e.g. A plan having an error usually means that we did not succeed at planning with unresolved parameters and therefore created a bogus plan with a high cost to make the planner prefer a custom plan. Unfortunately, if the current command is executed from a SQL function then custom plan logic is ignored and we might still try to execute the bogus plan. In this case, we throw the error that was generated during planning.